04-store-api/starter/controllers/products.js

```diff
@@ -0,0 +1,92 @@
+const Product = require('../models/product')
+
+const getAllProductsStatic = async (req, res) => {
+    // just throw an error, no need to use try-catch
+    // throw new Error('testing async erros for static products')
+
+    //const products = await Product.find( { featured: true, company: 'ikea' })
+
+    // using regex as search
+    const products = await Product.find({
+        price: { $gt: 30 }
+    }).sort('-price').select('name price').skip(1).limit(15)
+    res.status(200).json({ total: products.length, items: products })
+}
+
+
+const getAllProducts = async (req, res) => {
```

Comment:
Can you think any way that a malicious request might be able to do something bad here to make your server do more work than you might expect? Especially think about a case where there are a lot of products in the database.

Reply:
I think it has something to do with the regex expression and the fact we are not setting any limits for using it. I've checked the documentation and it looks like using $text operator should provide the same functionality.